core/domain/user_services.py

```diff
@@ -1412,8 +1413,26 @@ def update_profile_picture_data_url(
         profile_picture_data_url: str. New profile picture url to be set.
     """
     user_settings = get_user_settings(user_id, strict=True)
-    user_settings.profile_picture_data_url = profile_picture_data_url
-    save_user_settings(user_settings)
+    username = user_settings.username
+    if username is None:
+        raise utils.ValidationError(
+            'User does not have a valid username, having value None.')
+    # Ruling out the possibility of different types for mypy type checking.
+    assert isinstance(username, str)
```

Comment:
Does mypy fail if you omit this? I thought the `if username is None` block above would then force it to be str. /cc @sahiljoster32

Reply:
Oh it works without it, I guess I added this part before adding the `if username is None` check. Thanks for pointing it out.

Reply:
Seems to me that this conversation is resolved 🙂, but if not I'm happy to help!